### PR TITLE
Add confirmation to releasing when merge warning appears

### PR DESCRIPTION
**Pull request by Ge0rges**

### CODE REVIEW

**Comment by mwkaicz** [comment]
Why is that function string when string is never returned?
What it will do when you only left board of the craft without any command?

**Comment by Ge0rges** [comment]
That function hasn't been modified, also it returns ret.
The current functionality if for when you leave the craft without doing release.

**Comment by mwkaicz** [comment]
Sorry for that note about string, I overlooked that on cellphone.

But there is still problem, you can't block release task forever, ... f.e: what it will do when player leave board of the craft and then leave server too? It will be locked for that (offline) player until server restart ... same with world change, etc. block it until player call command is wrong. 

And I'm also not sure why you are using onAsyncPlayerChat and creating new listener for that, when there is capture of command `/release` which is doing the same thing. It should be better just put waitingForReleaseConfirmation=false inside it ... now (how do you wrote it) you will capture each chat message on server, without any real reason ... it will slow down server unnecessarily.

And finally: there is no relation between waitingForReleaseConfirmation and specific player ... one player leave the craft, another wrote release and it is in *** ... in case of it will works, because this can't work, you haven't set craft (c)    in ... CraftManager.getInstance().removeCraft( c );   

... please don't make requests without tested code and without tested functions in minecraft.

**Comment by Ge0rges** [comment]
Thanks for noticing that bug! I thought of looking for ./release but I fear since users don't have the permission to it, it might not get through. 

I fixed the public variable, I meant to change it but I guess I forgot to, sloppy of me.

You'll have to excuse my newsiness concerning Movecraft, I'll learn toon enough though!

**Comment by msummers123** [comment]
One more quick thing before I get back to work: If I do a video announcing all these awesome new changes, how do you want your name to be credited in the video? IE: what name should I use?

**Comment by Ge0rges** [comment]
You can credit me as Ge0rges (Gi0K) whichever you prefer, i have no preferences.

Thanks

**Comment by mwkaicz** [comment]
@msummers123 **be carefull with merging and releasing this, not all of my comments were fixed!** 

Still using AsyncChat without any important reason, still isn't fixed what happend when player left the world or the server, etc.  ... **Without more tests and validations "what happend if player do ... " I still consider this code as the threat for servers.**  Also I do not find as lucky way to kill release timeout forever.

**Comment by msummers123** [comment]
Actually for some reason I honestly don't understand very well, this and several other changes were not actually merged. I went looking for this code and it wasn't there. Github can be mysterious sometimes. Here is how I am thinking we will do this:
Make it so if it detects a potential merge, it will not automatically release the craft

...and that's it. It already forces the release of a player's craft when they leave the server, and you can already release by using a release sign or /release.

**Comment by msummers123** [comment]
Oh, and I'll make /manoverboard have some limitations so people can't use the craft that remains piloted to teleport across the map or between worlds.

**Comment by mwkaicz** [comment]
Maybe I don't understand too, so correct me, if I'm wrong. 

By the current code, it will prevent automatic release, when player levave the craft, until he write "release", right? .... So what happend when he don't write anything for many hours and will be still on server?  He will have still locked the craft after S hours? What can others players or admin do, when he is blocking some other structure/craft with his craft?

At this point I'm also not sure what happend when he leave world, but will be still on the server? I don't know if we have some event for this, but think that we haven't ... so player will not release some craft and go to the f.e. nether or another world via multiverse ... craft will be still prevented and locked for the player? Why?

Why must be there onAsyncPlayerChat for release confirmation, when we have command /release which makes same thing? ... it will capture each written message in vain.

Why there isn't any option to disable this function? I prefere low RAM usage instead of defending noobs, who don't know how to park their crafts. If 30 seconds isn't enough for noobs, increase timer, but don't replace it with necessity to write some command.

**Comment by msummers123** [comment]
Yes, if it thinks the craft may merge, it will prevent the craft from releasing. If the craft is not in a position that it will merge, it will automatically release. So this will only happen if the craft is badly parked. A piloted craft doesn't consume significant resources on the server, we are talking just a few K of memory. It would be the same if the pilot was just sitting in the craft not moving. So if a player could grief using this new feature, they could already grief just by sitting there. 

Other players can't force the craft to be released (unless they sink it). An admin could either use the new /release <player> to force the release of that player, or they can just /kick the player, since when they leave the server their craft is automatically released.

There is no onAsyncPlayerChat anymore, there is no need for that, as you say.

**Comment by mwkaicz** [comment]
there is a lot of "if" for me :-/ ... we don't want to release crafts for another players, it's not fairly and also not optimal for self-sustaining server. We are trying to make servers much independent on admins presence and intervention as it is possible. Also automatic (scripted) crafts will have problems when will be called something like /release all. 

Please, if you want to keep this in code, add there option to the config.yml to prevent this function.

File: src/main/java/net/countercraft/movecraft/listener/PlayerListener.java [hunk 1]
```diff
@@ -241,10 +243,11 @@ public void onPlayerHit( EntityDamageByEntityEvent event ) {
 	
 	@EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
 	public void onAsyncPlayerChat(AsyncPlayerChatEvent event) {
```
Comment (mwkaicz) [inline src/main/java/net/countercraft/movecraft/listener/PlayerListener.java]: I still see there onAsyncPlayerChat

**Comment by msummers123** [comment]
I think what you are missing is that this code was not actually merged. So this code is not in Movecraft.

**Comment by mwkaicz** [comment]
How is that possible, I see it in merged: https://github.com/msummers123/Movecraft-3/pulse

**Comment by msummers123** [comment]
Now that is a good question! :)

I honestly don't know what has happened. The code has simply vanished. Github said the merge was successful, but it is gone. I think it was because I merged so many pull requests so quickly. But in my version of the code, I do not have these changes. And yet, it thinks I am up to date. 

In the future, I will have to perform merges more slowly, checking each time.

**Comment by mwkaicz** [comment]
The second good question will be what all was merged successfully? :)

**Pull request merged**